examples/ex18.cpp

```diff
@@ -0,0 +1,304 @@
+   args.AddOption(&t_final, "-tf", "--t-final",
+                  "Final time; start time is 0.");
+   args.AddOption(&dt, "-dt", "--time-step",
+                  "Time step. Positive number skips CFL timestep calculation.");
+   args.AddOption(&cfl, "-c", "--cfl-number",
+                  "CFL number for timestep calculation.");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.AddOption(&vis_steps, "-vs", "--visualization-steps",
+                  "Visualize every n-th timestep.");
+
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return 1;
+   }
+   args.PrintOptions(cout);
+
+   // 2. Read the mesh from the given mesh file. This example requires a 2D
+   //    periodic mesh, such as ../data/periodic-square.mesh.
+   Mesh mesh(mesh_file, 1, 1);
+   const int dim = mesh.Dimension();
+
+   MFEM_ASSERT(dim == 2, "Need a two-dimensional mesh for the problem definition");
+
+   // 3. Define the ODE solver used for time integration. Several explicit
+   //    Runge-Kutta methods are available.
+   ODESolver *ode_solver = NULL;
+   switch (ode_solver_type)
+   {
+      case 1: ode_solver = new ForwardEulerSolver; break;
+      case 2: ode_solver = new RK2Solver(1.0); break;
+      case 3: ode_solver = new RK3SSPSolver; break;
+      case 4: ode_solver = new RK4Solver; break;
+      case 6: ode_solver = new RK6Solver; break;
+      default:
+         cout << "Unknown ODE solver type: " << ode_solver_type << '\n';
+         return 3;
+   }
+
+   // 4. Refine the mesh to increase the resolution. In this example we do
+   //    'ref_levels' of uniform refinement, where 'ref_levels' is a
+   //    command-line parameter.
+   for (int lev = 0; lev < ref_levels; lev++)
+   {
+      mesh.UniformRefinement();
+   }
+
+   // 5. Define the discontinuous DG finite element space of the given
+   //    polynomial order on the refined mesh.
+   DG_FECollection fec(order, dim);
+   // Finite element space for a scalar (thermodynamic quantity)
+   FiniteElementSpace fes(&mesh, &fec);
+   // Finite element space for a mesh-dim vector quantity (momentum)
+   FiniteElementSpace dfes(&mesh, &fec, dim, Ordering::byNODES);
+   // Finite element space for all variables together (total thermodynamic state)
+   FiniteElementSpace vfes(&mesh, &fec, num_equation, Ordering::byNODES);
+
+   // This example depends on this ordering of the space.
+   MFEM_ASSERT(fes.GetOrdering() == Ordering::byNODES, "");
+
+   cout << "Number of unknowns: " << vfes.GetVSize() << endl;
+
+   // 6. Define the initial conditions, save the corresponding mesh and grid
+   //    functions to a file. This can be opened with GLVis with the -gc option.
+   Array<int> offsets(num_equation + 1);
+   for (int k = 0; k <= num_equation; k++) { offsets[k] = k * vfes.GetNDofs(); }
```

Comment:
List the physical meaning of the components of `u_block`?

Reply:
Done in d767dc2.